reference/constraints/When.rst

```diff
@@ -0,0 +1,202 @@
+            <class name="App\Model\Discount">
+                <property name="value">
+                    <constraint name="GreaterThan">
+                        0
+                    </constraint>
+                    <constraint name="When">
+                        <option name="expression">
+                            this.type == 'percent'
+                        </option>
+                        <option name="constraints">
+                            <constraint name="LessThan">
+                                <option name="value">100</option>
+                                <option name="message">The value should be between 0 and 100!</option>
+                            </constraint>
+                        </option>
+                    </constraint>
+                </property>
+            </class>
+        </constraint-mapping>
+
+    .. code-block:: php
+
+        // src/Model/Discount.php
+        namespace App\Model;
+
+        use Symfony\Component\Validator\Constraints as Assert;
+        use Symfony\Component\Validator\Mapping\ClassMetadata;
+
+        class Discount
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint(new Assert\GreaterThan(0));
+                $metadata->addPropertyConstraint(new Assert\When([
+                    'expression' => 'this.type == "percent"',
+                    'constraints' => [
+                        new Assert\LessThan([
+                            'value' => 100,
+                        ]),
+                    ],
+                ]));
+            }
+
+            // ...
+        }
+
+The :ref:`expression <reference-constraint-expression-option>` option is the
+expression that must return true in order to trigger the validation of the attached constraints.
+To learn more about the expression language syntax, see
+:doc:`/components/expression_language/syntax`.
+
+For more information about the expression and what variables are available
+to you, see the :ref:`expression <reference-constraint-expression-option>`
+option details below.
+
+Options
+-------
+
+.. _reference-constraint-expression-option:
+
+``expression``
+~~~~~~~~~~~~~~
+
+**type**: ``string``
+
+The condition written with the expression language syntax that will be evaluated.
+If the expression evaluates to a false value (using ``==``, not ``===``),
+validation of constraints won't be triggered.
+
+To learn more about the expression language syntax, see
+:doc:`/components/expression_language/syntax`.
+
+Inside of the expression, you have access to up to 2 variables:
+
+Depending on how you use the constraint, you have access to 1 or 2 variables
+in your expression:
+
+* ``this``: The object being validated (e.g. an instance of Discount);
+* ``value``: The value of the property being validated (only available when
```

Comment:
could be great to give a code example using the `value` :)

Reply:
Done.